### PR TITLE
Rebuilt view "RecordPopover"

### PR DESCRIPTION
**Pull request by TomaszQr**
## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code is written according to the guidelines found [here] (https://yetiforce.com/en/knowledge-base/documentation/developer-documentation/item/how-to-create-php-files).
- [ ] I have signed the Contributor Licence Agreement between me and YetiForce.

### CODE REVIEW

File: modules/OSSTimeControl/models/RecordPopover.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		}
+		return $summaryFields;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getHeaderLinks(): array
+	{
+		$links = parent::getHeaderLinks();
```
Comment (rskrzypczak) [inline modules/OSSTimeControl/models/RecordPopover.php]:
```suggestion
		$linksModels = parent::getHeaderLinks();
```
$links = [];

File: modules/Calendar/models/RecordPopover.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		return $summaryFields;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getHeaderLinks(): array
+	{
+		$links = parent::getHeaderLinks();
```
Comment (rskrzypczak) [inline modules/Calendar/models/RecordPopover.php]:
```suggestion
		$linksModels = parent::getHeaderLinks();
```
$links = [];

File: modules/Calendar/models/RecordPopover.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			$links[] = [
+				'linktype' => 'RECORD_POPOVER_VIEW',
+				'linklabel' => 'DetailView',
+				'linkhref' => true,
+				'linkurl' => $detailUrl,
+				'linkicon' => 'fas fa-th-list',
+				'linkclass' => 'btn-sm btn-outline-secondary js-calendar-popover__button',
+			];
+		}
+		$linksModels = [];
```
Comment (rskrzypczak) [inline modules/Calendar/models/RecordPopover.php]: remove

File: modules/OSSTimeControl/models/RecordPopover.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			$links[] = [
+				'linktype' => 'RECORD_POPOVER_VIEW',
+				'linklabel' => 'DetailView',
+				'linkhref' => true,
+				'linkurl' => $this->recordModel->getDetailViewUrl(),
+				'linkicon' => 'fas fa-th-list',
+				'linkclass' => 'btn-sm btn-outline-secondary',
+			];
+		}
+		$linksModels = [];
```
Comment (rskrzypczak) [inline modules/OSSTimeControl/models/RecordPopover.php]: remove

File: modules/Reservations/models/RecordPopover.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ * Class Reservations_RecordPopover_Model.
+ */
+class Reservations_RecordPopover_Model extends Vtiger_RecordPopover_Model
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getHeaderLinks(): array
+	{
+		$links = parent::getHeaderLinks();
```
Comment (rskrzypczak) [inline modules/Reservations/models/RecordPopover.php]:
```suggestion
		$linksModels = parent::getHeaderLinks();
```
$links = [];

File: modules/Reservations/models/RecordPopover.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+			$links[] = [
+				'linktype' => 'RECORD_POPOVER_VIEW',
+				'linklabel' => 'DetailView',
+				'linkhref' => true,
+				'linkurl' => $this->recordModel->getDetailViewUrl(),
+				'linkicon' => 'fas fa-th-list',
+				'linkclass' => 'btn-sm btn-outline-secondary',
+			];
+		}
+		$linksModels = [];
```
Comment (rskrzypczak) [inline modules/Reservations/models/RecordPopover.php]: remove

File: modules/Vtiger/models/RecordPopover.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+		foreach ($this->recordModel->getModule()->getFields() as $fieldName => &$fieldModel) {
+			if ($fieldModel->isSummaryField() && $fieldModel->isViewableInDetailView() && !$this->recordModel->isEmpty($fieldName)) {
+				$summaryFields[$fieldName] = $fieldModel;
+			}
+		}
+		return $summaryFields;
+	}
+
+	/**
+	 * @return array
```
Comment (rskrzypczak) [inline modules/Vtiger/models/RecordPopover.php]: Add description.

File: modules/Vtiger/models/RecordPopover.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+	 */
+	public function getHeaderLinks(): array
+	{
+		return [];
+	}
+
+	/**
+	 * Returns list of fields to display.
+	 *
+	 * @return array
```
Comment (rskrzypczak) [inline modules/Vtiger/models/RecordPopover.php]: Vtiger_Field_Model[] ?

File: modules/Vtiger/models/RecordPopover.php [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+	 * Function to get model of view "RecordPopover".
+	 *
+	 * @param string               $moduleName
+	 * @param \Vtiger_Record_Model $recordModel
+	 *
+	 * @throws \App\Exceptions\AppException
+	 *
+	 * @return \Vtiger_RecordPopover_Model
+	 */
+	public static function getInstance(string $moduleName, Vtiger_Record_Model $recordModel): self
```
Comment (rskrzypczak) [inline modules/Vtiger/models/RecordPopover.php]: Vtiger_Record_Model $recordModel ?

File: modules/OSSTimeControl/models/RecordPopover.php [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getFields(): array
+	{
+		$summaryFields = [];
+		$fields = $this->getFieldsIcon();
+		$fieldsModel = $this->recordModel->getModule()->getFields();
+		foreach ($fields as $fieldName => $icon) {
```
Comment (mariuszkrzaczkowski) [inline modules/OSSTimeControl/models/RecordPopover.php]: foreach ($this->getFieldsIcon() as $fieldName => $icon) {

File: modules/Vtiger/models/RecordPopover.php [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+ */
+class Vtiger_RecordPopover_Model extends \App\Base
+{
+	/**
+	 * @var Vtiger_Record_Model
+	 */
+	protected $recordModel;
+
+	/**
+	 * @var string
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/models/RecordPopover.php]: no description

File: modules/Vtiger/models/RecordPopover.php [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+ * @author    Tomasz Kur <t.kur@yetiforce.com>
+ */
+
+/**
+ * Class Vtiger_RecordPopover_Model.
+ */
+class Vtiger_RecordPopover_Model extends \App\Base
+{
+	/**
+	 * @var Vtiger_Record_Model
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/models/RecordPopover.php]: no description

**Pull request merged**